### PR TITLE
Use `from_blob` based transfer to NEML2

### PR DESCRIPTION
**Pull request by dschwen**
Refs #27553

### CODE REVIEW

File: modules/solid_mechanics/src/neml2/userobjects/MOOSEMaterialPropertyToNEML2.C [hunk 1]
```diff
@@ -74,10 +75,18 @@ MOOSEMaterialPropertyToNEML2<T, state>::MOOSEMaterialPropertyToNEML2(const Input
 }
 
 template <typename T, unsigned int state>
-torch::Tensor
-MOOSEMaterialPropertyToNEML2<T, state>::convertQpMOOSEData() const
+void
+MOOSEMaterialPropertyToNEML2<T, state>::insertIntoInput(neml2::LabeledVector & input) const
 {
-  return NEML2Utils::toNEML2<T>(_mat_prop[_qp]);
+  input.base_index_put_(_neml2_variable, neml2::Tensor(torch::from_blob(const_cast<double *>(_buffer.data()), {_buffer.size() / _item_size,_item_size}), 1));
```
Comment (dschwen) [inline modules/solid_mechanics/src/neml2/userobjects/MOOSEMaterialPropertyToNEML2.C:81]: This needs to permit an arbitrary number of tensor dimensions.

**Comment by hugary1995** [comment]
`from_blob` is definitely the way to go.

In addition to the current conflicts, it'd be ideal to wait for https://github.com/applied-material-modeling/neml2/pull/177, as it will save some tensor reshaping operations on the moose side.

**Comment by dschwen** [comment]
Rebasing this is not feasible anymore and it is probably superseeded by #28597.

**Comment by dschwen** [comment]
I'll wait for the NEML2 2.0 update and might revisit `from_blob`. It turned out to have a modest impact on speed. However, the relative speedup might be more significant given the expected NEML2 performance increases related to 2.0.

**Pull request closed without merging**